deb/debian/stackdriver-agent.init

```diff
@@ -199,48 +166,54 @@ case "$1" in
         ;;
     stop)
         log_daemon_msg "Stopping $DESC" "$NAME"
-        d_stop
+        stop
         case "$?" in
-            0|1) log_end_msg 0 ;;
-            2) log_end_msg 1 ;;
+            0|3) log_end_msg 0 ;;
+            *) log_end_msg 1 ;;
         esac
         ;;
     status)
-        status_of_proc -p "$_PIDFILE" "$DAEMON" "$NAME" && exit 0 || exit $?
+        pidofproc -p "$_PIDFILE" "$DAEMON" >/dev/null
+        rc="$?"
+        case "$rc" in
+            0) log_success_msg "$NAME is running" ;;
+            4) log_failure_msg "could not access PID file for $NAME" ;;
+            *) log_failure_msg "$NAME is not running" ;;
+        esac
+        exit "$rc"
         ;;
     restart|force-reload)
         log_daemon_msg "Restarting $DESC" "$NAME"
-        check_config
-        rc="$?"
-        if test "$rc" -eq 1; then
-            log_progress_msg "not restarting, configuration error"
-            log_end_msg 1
-            exit 1
-        fi
-        d_stop
+        restart
         rc="$?"
         case "$rc" in
-            0|1)
-                sleep 1
-                d_start
-                rc2="$?"
-                case "$rc2" in
-                    0|1) log_end_msg 0 ;;
-                    2) log_end_msg 1 ;;
-                    3) log_end_msg 255; true ;;
-                    *) log_end_msg 1 ;;
-                esac
-                ;;
-            *)
-                log_end_msg 1
-                ;;
+            0) log_end_msg 0 ;;
+            1|2) log_end_msg 1 ;;
+            3) log_end_msg 255; true ;;
+            *) log_end_msg 1 ;;
         esac
+        exit "$rc"
         ;;
-    genhostid)
```

Comment:
If we remove `genhostid`, would this potentially break anything? It feels like a user visible change.

Reply:
Good to know.